File: android/app/src/main/java/com/tonhub/wallet/modules/store/EncryptionCallback.java
package com.tonhub.wallet.modules.store;

import com.facebook.react.bridge.Promise;

import org.json.JSONException;

import java.security.GeneralSecurityException;

import javax.crypto.Cipher;
import javax.crypto.spec.GCMParameterSpec;

public interface EncryptionCallback {
    Object run(
            Promise promise,
            Cipher cipher,
            PostEncryptionCallback postEncryptionCallback
    ) throws GeneralSecurityException, JSONException;
}

File: android/app/src/main/java/com/tonhub/wallet/modules/store/EncryptionCallback.kt
package com.tonhub.wallet.modules.store

import com.facebook.react.bridge.Promise
import org.json.JSONException
import java.security.GeneralSecurityException
import javax.crypto.Cipher

interface EncryptionCallback {
    @Throws(GeneralSecurityException::class, JSONException::class)
    fun run(
        promise: Promise,
        cipher: Cipher,
        postEncryptionCallback: PostEncryptionCallback
    ): Any?
}


